tests/integration/cmd_debug_test.go

```diff
@@ -0,0 +1,77 @@
+package integration
+
+import (
+	"os"
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/openshift/odo/tests/helper"
+)
+
+var _ = Describe("odo debug command tests", func() {
+	var project string
+	var context string
+
+	// Setup up state for each test spec
+	// create new project (not set as active) and new context directory for each test spec
+	// This is before every spec (It)
+	BeforeEach(func() {
+		SetDefaultEventuallyTimeout(10 * time.Minute)
+		SetDefaultConsistentlyDuration(30 * time.Second)
+		context = helper.CreateNewContext()
+		project = helper.CreateRandProject()
+		os.Setenv("GLOBALODOCONFIG", filepath.Join(context, "config.yaml"))
+	})
+
+	// Clean up after the test
+	// This is run after every Spec (It)
+	AfterEach(func() {
+		helper.DeleteProject(project)
+		helper.DeleteDir(context)
+		os.Unsetenv("GLOBALODOCONFIG")
+	})
+
+	Context("odo debug on a nodejs component", func() {
+		It("should expect a ws connection when tried to connect on different debug port locally", func() {
+			helper.CopyExample(filepath.Join("source", "nodejs"), context)
+			helper.CmdShouldPass("odo", "component", "create", "nodejs", "--project", project, "--context", context)
+			helper.CmdShouldPass("odo", "push", "--context", context)
+			go func() {
+				helper.CmdShouldRunWithTimeout(60*time.Second, "odo", "debug", "port-forward", "--local-port", "5050", "--context", context)
+			}()
+
+			// debug port
+			helper.HttpWaitForWithStatus("http://localhost:5050", "WebSockets request was expected", 12, 5, 400)
+		})
+
+		It("should expect a ws connection when tried to connect on default debug port locally", func() {
+			helper.CopyExample(filepath.Join("source", "nodejs"), context)
+			helper.CmdShouldPass("odo", "component", "create", "nodejs", "--project", project, "--context", context)
+			helper.CmdShouldPass("odo", "push", "--context", context)
+			go func() {
+				helper.CmdShouldRunWithTimeout(60*time.Second, "odo", "debug", "port-forward", "--context", context)
+			}()
+
+			// debug port
+			helper.HttpWaitForWithStatus("http://localhost:5858", "WebSockets request was expected", 12, 5, 400)
+		})
+
+		It("should expect a ws connection when used a different remote port", func() {
+			helper.CopyExample(filepath.Join("source", "nodejs"), context)
+			helper.CmdShouldPass("odo", "component", "create", "nodejs", "--project", project, "--context", context)
+			helper.CmdShouldPass("odo", "config", "set", "--force", "DebugPort", "9292", "--context", context)
+			dbgPort := helper.GetConfigValueWithContext("DebugPort", context)
+			Expect(dbgPort).To(Equal("9292"))
+			helper.CmdShouldPass("odo", "push", "--context", context)
+			go func() {
+				helper.CmdShouldRunWithTimeout(60*time.Second, "odo", "debug", "port-forward", "--context", context)
+			}()
+
+			// debug port
+			helper.HttpWaitForWithStatus("http://localhost:5858", "WebSockets request was expected", 12, 5, 400)
```

Comment:
```5858``` -> ```9292``` right ?

Reply:
no, because `DebugPort` only changes the remote port, not the local port

Reply:
please refer the debug proposal for more info https://github.com/openshift/odo/blob/master/docs/proposals/odo-debug.md#new-debugport-component-setting